### DISCUSSION TITLE
No emails in list with pilerimport

### DISCUSSION THREAD

**Discussion by alexis2m** [jsuto/piler#159] [General]
Hey,

After using pilerimport with IMAP, and having over 2600 received messages in the health stats. I cannot see any of them on the GUI logged as admin.

Do you have any ideas of that ?

It has been installed on Docker, and mariaDB shows emails in the metadata table.

Thanks

**Comment by jsuto** [comment]
Admin shouldn't see them. Login as an auditor or a regular user.

**Reply by alexis2m** [reply]
Okay perfect, when trying to log as an auditor too, no emails are being displayed.

I also tried creating an user account with the same email as the ones imported.
(It's for a personal archive)

**Reply by jsuto** [reply]
Show me the sphx query from the mail log when you hit the search button.

**Reply by alexis2m** [reply]
Where can I get those logs ?
In /var/log/mail.log, the file is empty

**Reply by jsuto** [reply]
It depends on your linux distro. Check your syslog server's config where it writes the mail facility logs.

**Reply by alexis2m** [reply]
Okay, then I have nothing in logs

**Reply by jsuto** [reply]
There must be something. Is your syslog daemon running? Does it log anything when you restart a piler daemon or login as a user on the UI?

**Reply by alexis2m** [reply]
Thanks a lot for your help, I found a way to get them.

Syslogs:
```
Aug 29 16:14:25 0f0a43d80b31 rsyslogd: [origin software="rsyslogd" swVersion="8.2001.0" x-pid="220" x-info="https://www.rsyslog.com"] start
Aug 29 16:14:25 0f0a43d80b31 piler-smtp[311]: reloaded config: /etc/piler/piler.conf
Aug 29 16:14:25 0f0a43d80b31 piler-smtp[311]: piler-smtp 1.3.8, build 998 starting
Aug 29 16:14:25 0f0a43d80b31 piler[313]: reloaded config: /etc/piler/piler.conf
Aug 29 16:14:25 0f0a43d80b31 piler[313]: piler 1.3.8, build 998 starting
Aug 29 16:15:01 0f0a43d80b31 CRON[330]: (piler) CMD (/usr/bin/find /var/piler/error -type f|wc -l > /var/piler/stat/error)
Aug 29 16:15:01 0f0a43d80b31 CRON[331]: (root) CMD (command -v debian-sa1 > /dev/null && debian-sa1 1 1)
Aug 29 16:15:01 0f0a43d80b31 CRON[332]: (piler) CMD (/usr/bin/indexer --config /etc/piler/sphinx.conf --quiet note1 --rotate)
Aug 29 16:15:01 0f0a43d80b31 CRON[333]: (piler) CMD (/usr/bin/find /var/piler/www/tmp -type f -name i.\* -exec rm -f {} \;)
Aug 29 16:15:01 0f0a43d80b31 CRON[334]: (piler) CMD (/usr/bin/indexer --config /etc/piler/sphinx.conf --quiet tag1 --rotate)
Aug 29 16:15:47 0f0a43d80b31 piler-webui[255]: sphinx query: 'SELECT id FROM main1,dailydelta1,delta1 WHERE        MATCH('(  maison ) &  (@from  | @to ) ') ORDER BY `sent` DESC LIMIT 0,20 OPTION max_matches=1000' in 0.01 s, 0 hits, 0 total found
```

mail logs
`Aug 29 16:15:47 0f0a43d80b31 piler-webui[255]: sphinx query: 'SELECT id FROM main1,dailydelta1,delta1 WHERE        MATCH('(  maison ) &  (@from  | @to ) ') ORDER BY `sent` DESC LIMIT 0,20 OPTION max_matches=1000' in 0.01 s, 0 hits, 0 total found`

Is it of any help ?

By the way, by checking the database, I found that everything is in order

**Comment by jsuto** [comment]
Yes, it helps. The problem is that the gui can't find your email address. Not sure what authentication method you use (IMAP authentication?), but you need to tell the gui what email address(es) you have. Also note that the admin user is NOT supposed to see the emails.

**Reply by alexis2m** [reply]
Yeah, actually it's a local install for saving my old personal emails / work emails before shutting it down.
Yeah, I'm now working using the Auditor account (auditor@local).
At one moment, emails were shown, but not anymore.
I did the import solely using pilerimport CLI.

**Comment by jsuto** [comment]
When using an auditor role account, then you should see all emails in the search results. Let me know if it's not the case.

**Reply by alexis2m** [reply]
Hi, thanks a lot for your time,
Sadly, using an Auditor account is not changing much as I don't see any emails, even when I do a basic research...

**Reply by jsuto** [reply]
Show me the sphinx query when you perform a search as an auditor. No search query, just hit the red search button.

**Reply by alexis2m** [reply]
Here is what I get:
`sphinx query: 'SELECT id FROM main1,dailydelta1,delta1 WHERE        MATCH('') ORDER BY `sent` DESC LIMIT 0,50 OPTION max_matches=1000' in 0.02 s, 0 hits, 0 total found`

**Comment by jsuto** [comment]
Your index seems to be empty. Run the following after ```mysql -h0 -P9306```

```show index main1 status;```
```show index dailydelta1 status;```

Also run ```select count(*) from sphx_index; ``` on the mysql piler database.

**Reply by alexis2m** [reply]
After trying to work it out I noticed the following:

- there is no table named main1, dailydelta1, or status
- my tables are sph_index, sph_counter and not sph**x**_index or sph**x**_counter

These tables are actually empty.
Meanwhile /var/piler/sphinx is full of `.delta` files

**Reply by jsuto** [reply]
OK, then show me "show tables" output.

**Reply by alexis2m** [reply]
Here are my tables:

```
+-------------------+
| Tables_in_piler   |
+-------------------+
| archiving_rule    |
| attachment        |
| audit             |
| autosearch        |
| counter           |
| counter_stats     |
| customer_settings |
| deleted           |
| domain            |
| domain_user       |
| email             |
| email_groups      |
| folder            |
| folder_extra      |
| folder_message    |
| folder_rule       |
| folder_user       |
| google            |
| google_imap       |
| group             |
| group_email       |
| group_user        |
| import            |
| ldap              |
| legal_hold        |
| metadata          |
| note              |
| online            |
| option            |
| private           |
| rcpt              |
| remote            |
| retention_rule    |
| search            |
| sph_counter       |
| sph_index         |
| sphx_counter      |
| sphx_index        |
| tag               |
| timestamp         |
| user              |
| user_settings     |
| v_attachment      |
| v_messages        |
+-------------------+
44 rows in set (0.001 sec)
```
Please note that I have created manually sphx_counter and sphx_index in an attempt to find a solution

**Comment by jsuto** [comment]
No, I meant the manticore tables.

**Reply by alexis2m** [reply]
Let me find that in my Docker installation

**Reply by alexis2m** [reply]
Hi, it appears that I don't have manticore on my Docker installation.
What's weird though it's that it worked for a couple of days first

**Comment by jsuto** [comment]
Can you try this docker compose file? https://github.com/jsuto/piler/blob/master/docker/docker-compose.yaml

**Reply by alexis2m** [reply]
Hi,
It's now working perfectly well.

I had some troubles installing the docker file (I'll make a pull request for useful changes / docs)

Do you have any steps to transfer old emails (from the original installation) to the new one as the email address does not exist anymore ?

Thanks for your help

**Comment by jsuto** [comment]
Awesome! To transfer legacy emails, then export them on the new host, copy the exported emails to new host, and then import them.

**Reply by alexis2m** [reply]
Okay, does transferring metadata / attachments and /etc/store is enough ?
I guess there is an automatic indexation tool ?

**Comment by jsuto** [comment]
I'd rather export then import to ensure everything is created just fine. Otherwise if you insist on your approach, then (assuming you have the same piler version on both hosts) copy the mysql piler database, the manticore index data, and everything in /var/piler/store, as well as the /etc/piler dir contents.

**Reply by alexis2m** [reply]
Okay thanks for your advice. I'll look to see how I'll be able to export emails from the original Piler instance that wasn't working as expected to the new instance working perfectly well
